### PR TITLE
Core: Fix JdbcCatalog CATALOG_TABLE_NAME to lowercase

### PR DESCRIPTION
**Pull request by haormj**
look https://github.com/apache/iceberg/issues/2744

### CODE REVIEW

**Comment by haormj** [comment]
@ismailsimsek please review :-)

**Comment by ismailsimsek** [comment]
looks good, do you want o add second fix too ? 
changing varchar size like below? 

```
      "CREATE TABLE " + CATALOG_TABLE_NAME +
          "(" +
          CATALOG_NAME + " VARCHAR(255) NOT NULL," +
          TABLE_NAMESPACE + " VARCHAR(255) NOT NULL," +
          TABLE_NAME + " VARCHAR(255) NOT NULL," +
          METADATA_LOCATION + " VARCHAR(5500)," +
          PREVIOUS_METADATA_LOCATION + " VARCHAR(5500)," +
          "PRIMARY KEY (" + CATALOG_NAME + ", " + TABLE_NAMESPACE + ", " + TABLE_NAME + ")" +
          ")";
```

**Comment by haormj** [comment]
@ismailsimsek of course :-)

**Review by XuQianJin-Stars — COMMENTED** [review]
+1

File: core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java [hunk 1]
```diff
@@ -27,7 +27,7 @@
 import org.apache.iceberg.relocated.com.google.common.collect.Iterables;
 
 final class JdbcUtil {
-  protected static final String CATALOG_TABLE_NAME = "ICEBERG_TABLES";
+  protected static final String CATALOG_TABLE_NAME = "iceberg_tables";
```
Comment (openinx) [inline core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java:30]: What's the specific reason that we need to change it to lower case  ?  Just want to align with the other table names ?  (I read the context from https://github.com/apache/iceberg/issues/2744 but still did not get the point).

File: core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java [hunk 2]
```diff
@@ -42,11 +42,11 @@
...
-          TABLE_NAMESPACE + " VARCHAR(1255) NOT NULL," +
-          TABLE_NAME + " VARCHAR(1255) NOT NULL," +
-          METADATA_LOCATION + " VARCHAR(32768)," +
-          PREVIOUS_METADATA_LOCATION + " VARCHAR(32768)," +
+          CATALOG_NAME + " VARCHAR(255) NOT NULL," +
+          TABLE_NAMESPACE + " VARCHAR(255) NOT NULL," +
+          TABLE_NAME + " VARCHAR(255) NOT NULL," +
+          METADATA_LOCATION + " VARCHAR(5500)," +
+          PREVIOUS_METADATA_LOCATION + " VARCHAR(5500)," +
           "PRIMARY KEY (" + CATALOG_NAME + ", " + TABLE_NAMESPACE + ", " + TABLE_NAME + ")" +
```
Comment (openinx) [inline core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java:50]: I think the reason to choose 255 for the size of `CATALOG_NAME`, `TABLE_NAMESPACE`, `TABLE_NAME` is :  we are composting the three into a primary key,  and the primary key size is limited by 767 ( 255 * 3 = 765 < 767).

Okay, that sounds good to me.

File: core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java (hunk 1 above)
Comment (XuQianJin-Stars) [reply core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java:30]: > What's the specific reason that we need to change it to lower case ? Just want to align with the other table names ? (I read the context from #2744 but still did not get the point).

PostgreSQL's database kernel is case sensitive, eg:  https://www.cnblogs.com/gaofan/p/11398753.html

Comment (openinx) [reply core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java:30]: OK,  got it!

**Comment by openinx** [comment]
Got this merged, thanks @haormj for contributing, and thanks @ismailsimsek @XuQianJin-Stars for reviewing !

**Pull request merged**